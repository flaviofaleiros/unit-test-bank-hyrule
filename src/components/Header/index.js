import React from 'react';
import PropTypes from 'prop-types';

const Header = ({ title }) => {
    return(
        <header className="App-header">
            <h1>{title}</h1>
        </header>
    );
};

Header.defaultProps = {
    title: 'Hyrule Bank',
}

Header.propTypes = {
    title: PropTypes.string,
};

export default Header;
