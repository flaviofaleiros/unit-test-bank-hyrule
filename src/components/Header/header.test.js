import React from "react";
import {render, screen } from "@testing-library/react";

import Header from "./index";

describe('render Header', () => {

    it('should render component', function () {
        render(<Header />);
        expect(screen.getByText('Hyrule Bank')).toBeInTheDocument();
    });

    it('should Render PagBank', function () {
        render(<Header title={'PagBank'}/>);
        expect(screen.getByText('PagBank')).toBeInTheDocument();
    });

    test('should snapShot', function () {
        const { container } = render(<Header title={'PagBank'}/>)
        expect(container.firstChild).toMatchSnapshot();
    });
});
