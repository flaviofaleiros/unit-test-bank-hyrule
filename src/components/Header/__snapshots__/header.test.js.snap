// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Header should snapShot 1`] = `
<header
  class="App-header"
>
  <h1>
    PagBank
  </h1>
</header>
`;
